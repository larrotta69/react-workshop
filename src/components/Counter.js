import React, { Component } from 'react'

export class Counter extends Component {

    state = {
        count: 0,
    };

    makeIncrement = amount => () =>
      this.setState({
          count: this.state.count + amount
      });


    increment = this.makeIncrement(1)

    render() {
      return (
          <div>
          <p>Current count: {this.state.count}</p>
          <button onClick={this.increment}>Increment count</button>
          </div>
      )
    }
}

export default Counter
