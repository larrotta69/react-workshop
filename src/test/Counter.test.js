import React from 'react'
import Counter from '../components/counter'

import { shallow, mount, render } from 'enzyme'
import renderer from 'react-test-renderer';

describe('Counter component', () => {

  it('Matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Start with a count of 0', () => {
    const wrapper = shallow(<Counter />)
    const text = wrapper.find('p').text()
    expect(text).toEqual('Current count: 0')
  })

  it('Can increment the count when the button is clicked', () => {
    const wrapper = shallow(<Counter />)
    const incrementBtn = wrapper.find('button')
    incrementBtn.simulate('click')
    const text = wrapper.find('p').text()
    expect(text).toEqual('Current count: 1')
  })
});
