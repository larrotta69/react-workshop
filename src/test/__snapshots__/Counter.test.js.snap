// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Matches the snapshot 1`] = `
<div>
  <p>
    Current count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment count
  </button>
</div>
`;
